lib/authentication/service/auth_provider.dart

```diff
@@ -184,6 +148,38 @@ class AuthProvider with ChangeNotifier {
     return _firebaseUser.uid;
   }
 
+  Future<void> _emigrate(Map<String,dynamic> data) async{
```

Comment:
Nit: It can simply return `void`.

Reply:
With out Future<void>, when you call the method a warring appears that suggest to make it future<void>